packages/react/collection/src/collection.tsx

```diff
@@ -59,14 +58,14 @@ function createCollection<E extends React.ElementRef<any> = void, S = {}>(name:
     ssrSyncUseCollectionItemCountRef.current = 0;
 
     return (
-      <CollectionContext.Provider
-        value={React.useMemo(
-          () => ({ items, addItem, removeItem, ssrSyncUseCollectionItemCountRef }),
-          [items, addItem, removeItem]
-        )}
+      <CollectionProviderImpl
+        items={items}
+        addItem={addItem}
+        removeItem={removeItem}
```

Comment:
I'll wait for #496 to merge and then update these because they should be `onItemAdd` and `onItemRemove` and then we can inline the handlers